packages/amplify-cli/src/init-steps/s0-analyzeProject.ts

```diff
@@ -220,6 +230,108 @@ function getDefaultEnv(context: $TSContext): string | undefined {
   return undefined;
 }
 
+async function setEnvVar(context: $TSContext, currentEnvName: string, projectPath: string) {
+  const teamProviderInfo = stateManager.getTeamProviderInfo(projectPath, {
+    throwIfNotExist: false,
+    default: {},
+  });
+  const functions = teamProviderInfo[currentEnvName]?.categories?.function || {};
+  const functionNames = Object.keys(functions);
+  if (functionNames.length === 0) {
+    return {};
+  }
+
+  const envVarQuestion = async () => {
+    const envVarQuestion: inquirer.ListQuestion = {
+      type: 'list',
+      name: 'envVar',
+      message: 'You have custom function environment variables configured. What do you want to do?',
+      choices: [
+        {
+          value: 'carry',
+          name: 'Carry existing values over to the new environment (values can be updated later using amplify update function)',
+        },
+        {
+          value: 'update',
+          name: 'Update values now',
+        },
+      ],
+    };
+    if (context.parameters.options.quickstart) return;
+    const { envVar } = await inquirer.prompt(envVarQuestion);
+    if (envVar === 'carry') return;
```

Comment:
how are the values in the new env being set in this case?

Reply:
Then it goes to the next line:
https://github.com/aws-amplify/amplify-cli/blob/ee3002458af30c7caad8197570d0ac92eb4d4b26/packages/amplify-cli/src/init-steps/s0-analyzeProject.ts#L263

We have default values already. When users want to update one of the env vars, then it follows the below steps:
envVarSelectFunction => envVarSelectKey => envVarUpdateValue